docs/spec.md

```diff
@@ -5658,24 +6055,26 @@ Performs element-wise XOR of two tensors `lhs` and `rhs` and produces a `result`
 tensor. Depending on the element type, does the following:
 
 * For booleans: logical XOR.
-* For integers: bitwise XOR.
+* For integers or quantized types: bitwise XOR.
 
 #### Inputs
 
-| Label | Name  | Type                              | Constraints |
-|-------|-------|-----------------------------------|-------------|
-| (I1)  | `lhs` | tensor of boolean or integer type | (C1)        |
-| (I2)  | `rhs` | tensor of boolean or integer type | (C1)        |
+| Label | Name  | Type                                         | Constraints |
+|-------|-------|----------------------------------------------|-------------|
+| (I1)  | `lhs` | tensor of boolean, integer or quantized type | (C1-C2)     |
+| (I2)  | `rhs` | tensor of boolean, integer or quantized type | (C1)        |
 
 #### Outputs
 
-| Name     | Type                              | Constraints |
-|----------|-----------------------------------|-------------|
-| `result` | tensor of boolean or integer type | (C1)        |
+| Name     | Type                                         | Constraints |
+|----------|----------------------------------------------|-------------|
+| `result` | tensor of boolean, integer or quantized type | (C1)        |
 
 #### Constraints
 
 * (C1) `lhs`, `rhs` and `result` have the same type.
+* If the operation uses quantized tensors:
+  * (C2) `is_empty(quantization_dimension(lhs))`.
 
 #### Examples
 
```

Comment:
For the integer-only ops, how about we just decline to support them for quantized tensors?

This way, we can describe the semantics of elementwise ops with respect to quantization as "we treat quantized tensors as floating-point tensors and wrap floating-point computations in dequantize/quantize".

If someone wants to do integer-only computations on quantized tensors, they can always bitcast_convert to the storage type.

Reply:
I do find it is confusing about integer-only computation and computation on quantized tensors. My understanding is that there are two ways to do computation on quantized tensors: 
1. dequant->float op->quant
2. integer-only computation with multiplier and shift (#1475)
Here `integer-only computation` is about implementation. 

From the spec perspective,  an `integer-only computation` op seems to mean that mathematically it can only act on integer operands. In that sense `we just decline to support them for quantized tensors` is rigorous. 

Let me know if my understanding is correct. If so, is there a way to make the distinction more clear?

Reply:
Apologies for the confusion! What I meant to discuss in this comment thread is the ops which don't support floating-point tensors, i.e.: `and`, `count_leading_zero`, `not`, `or`, `popcnt`, `shift_left`, `shift_right_arithmetic`, `shift_right_logical`, `xor`.

The PR description proposes to support quantized tensors in these ops and directly use integer representations of their elements, as if these elements were actually of integer type and not of quantized element type. I would propose something different - to not allow quantized tensors in these ops at all, and if someone wants to operate on integer representations directly, then they would need to bitcast_convert.

What would happen to quantized element types with multiplier and shift - that seemed to be out of scope of this PR, so I haven't thought about that yet :)

Reply:
sgtm. `bitcast_convert` is rigorous.

Reply:
Agree with the proposal. Also updated the PR description accordingly.